### PR TITLE
Fix undefined property

### PR DESCRIPTION
**Pull request by idnovate**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When you try to update cart from an Admin controller from a module it appears an error because customer doesn't exist in context
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | No
| Deprecations? | No

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11763)

### CODE REVIEW

File: classes/Cart.php [hunk 1]
```diff
@@ -1324,6 +1324,8 @@ public function updateQty(
                 // The $id_address_delivery must be linked with customer
                 $id_address_delivery = 0;
             }
+        } else {
+	        $id_address_delivery = 0;
```
Comment (Quetzacoalt91) [inline classes/Cart.php]: Why is it necessary to reset the $id_address_delivery value? I thought you wanted to cover a case when the variable does not exist, but it exist from the method signature.

Comment (idnovate) [reply classes/Cart.php]: I just wanted to catch the warning when `Context::getContext()->customer` doesn't exist! `$id_address_delivery = 0` is default value when customer doesn't has address.

My problem comes when I remove a cart rule with a free gift from the backoffice and I update cart.

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by Quetzacoalt91** [comment]
Sending this PR to QA review. In the meantime @idnovate, please take some time to run `./vendor/bin/php-cs-fixer fix --show-progress=dot` on your folder and commit the modified files in order to pass the lint jobs of Travis.

**Comment by marionf** [comment]
Hello @idnovate 

Can you send me the module ? It will alowed to reproduce the issue and check your PR

Thanks :)

**Comment by idnovate** [comment]
> Sending this PR to QA review. In the meantime @idnovate, please take some time to run `./vendor/bin/php-cs-fixer fix --show-progress=dot` on your folder and commit the modified files in order to pass the lint jobs of Travis.

I did a modification to the code and executed `php-cs-fixer`

**Comment by idnovate** [comment]
> Hello @idnovate
> 
> Can you send me the module ? It will alowed to reproduce the issue and check your PR
> 
> Thanks :)

I attach you a very simple module -> [dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/2850807/dummymodule.zip).

How to reproduce the issue:
- Install module
- Create a cart rule with a free gift as action. No conditions nor coupon is needed.
- In a new tab with **Incognito** mode, **no customer logged in**, add a product to the cart. Cart rule should be applied automatically. 
- Open module configuration and the loop to remove the cart rules is executed. The error appears:
image

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @idnovate

**Pull request merged**